java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java

```diff
@@ -363,6 +363,9 @@ private <RequestT> SdkHttpFullRequest prepareRequest(
             }
             req.putHeader("Content-Length", String.valueOf(body.getContentLength()));
             req.contentStreamProvider(body::getInputStream);
+            // To add the "X-Amz-Content-Sha256" header, it needs to set as required.
+            // It is a required header for Amazon OpenSearch Serverless.
+            req.putHeader("x-amz-content-sha256", "required");
```

Comment:
Yeah it took me a while to find too. We use `Aws4Signer` which does not have the sign method overriden like `Aws4UnsignedPayloadSigner`.  According to documentation, the `Aws4UnsignedPayloadSigner` is similar to `Aws4Signer` but just adds `UNSIGNED-PAYLOAD` when protocol is HTTPS. Can we use `Aws4UnsignedPayloadSigner`?

Reply:
If it works we sure can I think.

Reply:
I tried the `Aws4UnsignedPayloadSigner` but it does not work for Amazon OpenSearch Service since it signs the payload with `UNSIGNED_PAYLOAD` over https protocol. Works for Amazon OpenSearch Serverless. I can PR this change on their repo may be as a new signer class but for now looks like we might have to use the hard-coded "required".